doc/developer-guide/developer-guide.rst

```diff
@@ -0,0 +1,164 @@
+the Dome Seeing Monitor UI and sends that information across SAL for recording
+in the EFD. The CSC also has two simulation modes in addition to the standard
+operational (real) mode and they will be described later. 
+
+.. _Dependencies:
+
+Dependencies
+============
+
+* `SAL <https://ts-sal.lsst.io>`_ - 5.0.0
+* ts_xml - v6.1.0
+* ts_salobj - v6.0.0
+* `aionotify <https://github.com/rbarrois/aionotify>`_ - 0.2.0
+
+.. Linking to the previous versions may also be worthwhile, depending on the CSC
+
+.. _API:
+
+DSM API
+=============================
+
+The content in this section is autogenerated from docstrings.
+
+.. The code below should insert the docstrings from the code.
+
+.. automodapi:: lsst.ts.dsm
+    :no-main-docstr:
+    :no-inheritance-diagram:
+
+
+.. _Build:
+
+Build and Test
+==============
+
+The recommended method for development is to get the T&S development Docker container
+(need doc link). Since the DSM code is not shipped with the development container, a
+clone of the code is necessary. Determine a place on your computer and get it via this
+
+.. prompt:: bash
+
+  git clone git@github.com:lsst-ts/ts_dsm.git
+
+It is recommended that the code be cloned into another directory as the potential
+of needing other repositories is possible. Change or create the appropriate branch in the clone. Remember the location housing the repository.
+
+At the time of this writing, the development container to retrieve is ``lsstts/develop-env:c0010``. Note that it is NOT compatible with SAL 5 and SALOBJ 6 out of the 
+box. Those packages must be cloned, ``eups declare`` run and branch switched before
+embarking on the development of the ``ts_dsm`` code. The OpenSplice configuration
+must also be changed to handle the new software versions. This will not be covered as the
+hope is a new development container will soon appear.
+
+To build the code and run the unit tests, do the following.
+
+.. code-block:: bash
+
+    docker run -it --name {name for container} -v {repository_location}:/home/saluser/develop lsstts/develop-env:c0010
+    cd develop/ts_dsm
+    setup -kr .
+    pip install aionotify
+    scons
+
+The ``pip`` installation of ``aionotify`` is required since the package does not come
+with the development environment. While developing the code, ``scons`` can be used to run the unit tests after code changes.
+
+.. _Usage:
+
+Usage
+=====
+
+There are two scripts that perform operations of the CSC. They are ``run_dsm.py`` and ``shutdown_dsm.py``. The parameters they take can be found by passing ``-h`` or
+``--help`` to the given script. The ``run_dsm.py`` script constructs the CSC and sends
+it to ``ENABLED`` state, allowing for the CSC to function straight away. It can be
+run in real mode or one of two simulation modes. The simulation modes will be shown
+in the next session. To run the CSC in real mode, do the following.
+
+.. prompt:: bash
+
+  run_dsm.py
```

Comment:
I strongly suggest that "how to run the DSM", "how to run it in simulation mode" and "how to configure" be part of the User Guide. 

My mental model is:
* User: somebody interacts with, runs or stops the CSC.
* Developer: somebody who modifies the code.

Reply:
This type of suggestion needs to be run back up to the documentation effort. It's a fundamental shift in how the docs are organized and the boilerplate structure does not reflect this.

Reply:
For what it's worth: I already received permission to do it that way for my docs and am writing all my CSC docs that way. So a good fraction of our CSCs will be documented that way.